### PR TITLE
`NoWrap` `Text` feature

### PR DESCRIPTION
**Pull request by ickshonpe**
# Objective

In Bevy 10.1 and before, the only way to enable text wrapping was to set a local `Val::Px` width constraint on the text node itself. `Val::Percent` constraints and constraints on the text node's ancestors did nothing.

#7779 fixed those problems. But perversely displaying unwrapped text is really difficult now, and requires users to nest each `TextBundle` in a `NodeBundle` and apply `min_width` and `max_width` constraints. Some constructions may even need more than one layer of nesting. I've seen several people already who have really struggled with this when porting their projects to main in advance of 0.11.

## Solution

Add a `NoWrap` variant to the `BreakLineOn` enum.
If `NoWrap` is set, ignore any constraints on the width for the text and call `TextPipeline::queue_text` with a width bound of `f32::INFINITY`.

---

## Changelog
* Added a `NoWrap` variant to the `BreakLineOn` enum.
* If `NoWrap` is set, any constraints on the width for the text are ignored and `TextPipeline::queue_text` is called with a width bound of `f32::INFINITY`.
* Changed the `size` field of `FixedMeasure`  to `pub`. This shouldn't have been private, it was always intended to have `pub` visibility.
* Added a `with_no_wrap` method to `TextBundle`.

## Migration Guide

`bevy_text::text::BreakLineOn` has a new variant `NoWrap` that disables text wrapping for the `Text`.
Text wrapping can also be disabled using the `with_no_wrap` method of `TextBundle`.

### CODE REVIEW

**Review by mockersf — APPROVED** [review]

File: crates/bevy_text/src/text.rs [hunk 1]
```diff
@@ -186,12 +186,17 @@ pub enum BreakLineOn {
...
     AnyCharacter,
+    /// Lines will never be broken up.
+    NoWrap,
 }
 
 impl From<BreakLineOn> for glyph_brush_layout::BuiltInLineBreaker {
     fn from(val: BreakLineOn) -> Self {
         match val {
-            BreakLineOn::WordBoundary => glyph_brush_layout::BuiltInLineBreaker::UnicodeLineBreaker,
+            // If `NoWrap` is set the text is given unbounded width, so the `BuiltInLineBreaker` won't be used.
```
Comment (nicoburns) [inline crates/bevy_text/src/text.rs:196]: I think this comment isn't quite accurate, in that I would expect wrapping to still occur if the text contains an explicit newline character. It's just that soft wrapping won't occur at "line-break opportunities".

File: crates/bevy_ui/src/widget/text.rs [hunk 2]
```diff
@@ -91,7 +91,13 @@ fn create_text_measure(
         text.linebreak_behavior,
     ) {
         Ok(measure) => {
-            content_size.set(TextMeasure { info: measure });
+            if matches!(text.linebreak_behavior, BreakLineOn::NoWrap) {
```
Comment (nicoburns) [inline crates/bevy_ui/src/widget/text.rs:94]: Could just use `==` here (or `match` instead of `if`).

**Review by nicoburns — APPROVED** [review]
Minor style/documentation changes requested, but the meat of the implementation looks good to me.

File: crates/bevy_text/src/text.rs (hunk 1 above)
Comment (ickshonpe) [reply crates/bevy_text/src/text.rs:196]: I'm not sure, as the line breaker is only used for soft breaks I think so maybe it's correct?

We need to be very clear that `NoWrap` only applies to soft breaks and not hard breaks caused by a newline though definitely.

Comment (ickshonpe) [reply crates/bevy_text/src/text.rs:196]: No I'm wrong, with either enum internally it uses a `xi_unicode::LineBreakIterator` that handles both hard and soft breaks.

**Review by StaffEngineer — APPROVED** [review]

**Review by nicoburns — APPROVED** [review]
My nits have now been addressed, so I think this is good to go :)

**Review by alice-i-cecile — APPROVED** [review]

**Pull request merged**